src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -52,31 +52,63 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
-                                    types: tp.ConstraintTypes.Select(t => syntaxGenerator.TypeExpression(t)));
-                            }
-                        }
-                    }
+            return syntaxGenerator.Declaration(symbol);
+        }
+
+        // Temporary solution till corresponding Roslyn API is added: https://github.com/dotnet/arcade/issues/11895.
+        private static SyntaxNode ExplicitInterfaceImplementationMethodDeclaration(this SyntaxGenerator syntaxGenerator, IMethodSymbol method, string name, IEnumerable<SyntaxNode>? statements = null)
+        {
+            var decl = syntaxGenerator.MethodDeclaration(
+                name,
+                parameters: method.Parameters.Select(p => syntaxGenerator.ParameterDeclaration(p)),
+                returnType: method.ReturnType?.SpecialType == SpecialType.System_Void ? null : syntaxGenerator.TypeExpression(method.ReturnType!),
+                accessibility: method.DeclaredAccessibility,
+                modifiers: DeclarationModifiers.From(method),
+                statements: statements);
 
-                    return declaration;
+            if (method.TypeParameters.Length > 0)
+            {
+                decl = syntaxGenerator.WithTypeParametersAndConstraintsCopyExt(decl, method.TypeParameters);
+            }
+
+            return decl;
+        }
+
+        // this is copy/paste from private method `SyntaxGenerator.WithTypeParametersAndConstraints`
+        private static SyntaxNode WithTypeParametersAndConstraintsCopyExt(this SyntaxGenerator syntaxGenerator, SyntaxNode declaration, ImmutableArray<ITypeParameterSymbol> typeParameters)
+        {
+            if (typeParameters.Length > 0)
+            {
+                declaration = syntaxGenerator.WithTypeParameters(declaration, typeParameters.Select(tp => tp.Name));
+
+                foreach (var tp in typeParameters)
+                {
+                    if (tp.HasConstructorConstraint || tp.HasReferenceTypeConstraint || tp.HasValueTypeConstraint || tp.ConstraintTypes.Length > 0)
+                    {
+                        declaration = syntaxGenerator.WithTypeConstraint(declaration, tp.Name,
+                            kinds: (tp.HasConstructorConstraint ? SpecialTypeConstraintKind.Constructor : SpecialTypeConstraintKind.None)
+                                   | (tp.HasReferenceTypeConstraint ? SpecialTypeConstraintKind.ReferenceType : SpecialTypeConstraintKind.None)
+                                   | (tp.HasValueTypeConstraint ? SpecialTypeConstraintKind.ValueType : SpecialTypeConstraintKind.None),
+                            types: tp.ConstraintTypes.Select(t => syntaxGenerator.TypeExpression(t)));
+                    }
                 }
             }
 
-            return syntaxGenerator.Declaration(symbol);
+            return declaration;
```

Comment:
Why are we returning the SyntaxNode declaration when it's also an input parameter so it's changed anyway?

Reply:
It's like override of original `SyntaxGenerator.Declaration` methods for several use cases. If the use cases are not covered - call original `Declaration` method.